processing/src/main/java/org/apache/carbondata/processing/sortandgroupby/sortdata/IntermediateFileMerger.java

```diff
@@ -116,8 +116,15 @@ public IntermediateFileMerger(SortParameters mergerParameters, File[] intermedia
           writeDataTofile(next());
         }
       } else {
+        int i = 0;
         while (hasNext()) {
+          i++;
           writeDataTofileWithOutKettle(next());
+          if (i % 10000 == 0) {
```

Comment:
it is unnecessary calculation we are adding in it. Already the buffer size is configurable so once buffer size reaches buffered out stream can flush it down.